.github/workflows/tests.yml

```diff
@@ -0,0 +1,87 @@
+name: tests
+on:
+  schedule:
+    # build at least once per week
+    - cron: "0 0 * * 0"
+  pull_request:
+  push:
+    paths-ignore:
+      - "README.md"
+      - "CHANGELOG.md"
+jobs:
+  # TODO: enable after running cargo fmt again
+  # check:
+  #   name: Check formatting
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #     - uses: actions/checkout@v2
+  #     - uses: actions-rs/toolchain@v1
+  #       with:
+  #         toolchain: stable
+  #         override: true
+  #         components: rustfmt
+  #         profile: minimal
+  #     - name: Check formatting
+  #       uses: actions-rs/cargo@v1
+  #       with:
+  #         command: fmt
+  #         args: --all -- --check
+  test:
+    name: Test
+    runs-on: ${{ matrix.os }}
+    env:
+      RUST_BACKTRACE: 1
+    strategy:
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
```

Comment:
that's a whole lot of build: 3 * 2 * 2 in total. Can we perhaps mix and match those into fewer builds?

Reply:
I was actually considering adding _more_ builds by adding all of the different permutations of features (i.e. none, mint, serde, mint+serde). This is what the strategy matrix is for :)

I could certainly remove the windows and macos builds, but is it actually an improvement? Yes there are `3 * 2 * 2` total builds here, but they run in parallel and the entire CI process finishes in under 3 minutes. I don't consider this to be excessive. I'd much prefer to run CI on all the major platforms and only reduce this if there's an issue with long build queues.